feat(recap): Enhances appellate docket purchase to support ACMS cases

ERosendo opened the pull request:
Key changes: 

- Introduces a new helper function that centralizes logic to determine whether a given `court_id` requires ACMS-specific handling

- Updates `purchase_appellate_docket_by_docket_number` to detect ACMS cases and switch to `ACMSDocketReport` when appropriate. Falls back to `AppellateDocketReport` for standard PACER courts.

- Tweaks `merge_pacer_docket_into_cl_docket` to save raw ACMS data as `.json` instead of `.html` (**ACMS** does not return a traditional **HTML** response)

- Adds `FakeAcmsCaseSearch`, `FakeAcmsDocketReport`, and related fakes for isolated ACMS test behavior.


Depends on #5938 and https://github.com/freelawproject/juriscraper/pull/1494

Fixes #5154

ERosendo commented on cl/recap/tasks.py:
```diff
@@ -2420,12 +2422,39 @@ def purchase_appellate_docket_by_docket_number(
         raise ParsingException("No data found in docket report.")
+
+    if acms_case_id:
+        docket_data["docket_entries"] = sorted(
+            docket_data["docket_entries"],
+            key=lambda d: (
+                d["date_filed"],
+                d["document_number"] is None,
+                d["document_number"],
+            ),
+        )
```
You're correct. I'll add the new helper method and refactor this code

albertisfu replied:
btw when testing the fetch attachment page PR I found this docket:
https://www.courtlistener.com/docket/68490335/united-states-of-america-v-raji/

Which I retrieved using the fetch API in my local env.

As you can see the entry 17 is in the wrong place due to date Field is being prioritized over the document number.

<img width="1156" height="295" alt="Screenshot 2025-07-16 at 2 45 32 p m" src="https://github.com/user-attachments/assets/d778ea13-ee76-458c-ae7f-6e24f6e6a672" />

I checked in PACER and the date filed is correct.

Would it be possible to prioritize the document number, and use the date filed if it's None?

ERosendo replied:
Since we just introduced a method to sort ACM entries, I think we could check whether all docket entries in the input have a document_number. If they do, we can sort purely by that number. Otherwise, we fall back to the existing sort function. What do you think?

ERosendo replied:
cool! i'll refine the new helper method

ERosendo commented:
@albertisfu Thanks for the review. I’ve addressed all your comments and also added a final refinement to the logic for sorting ACMS entries.

> However, I’m curious where get_acms_tokens is being set to True to trigger ACMS authentication. I couldn’t find it in the code.

Currently, the token is generated only when a user attempts to fetch an ACMS document. I’m planning to open a follow-up PR to store the ACMS credentials in cache so we can reuse them and avoid recomputing the token..

albertisfu reviewed (approved):
Thanks @ERosendo this looks ready to be merged!
